unix-ffi/unittest/unittest.py

```diff
@@ -490,19 +498,22 @@ def main(module="__main__", testRunner=None):
     else:
         runner = TestRunner()
 
-    if len(sys.argv) <= 1:
-        result = discover(runner)
-    elif sys.argv[0].split(".")[0] == "unittest" and sys.argv[1] == "discover":
+    argn = len(sys.argv)
+    # See https://docs.python.org/3/library/unittest.html#test-discovery:
+    # python -m unittest is the equivalent of python -m unittest discover.
+    if not argn or (
+        sys.argv[0].split(".")[0] == "unittest"
+        and (argn == 1 or (argn >= 1 and sys.argv[1] == "discover"))
+    ):
         result = discover(runner)
+    # If there's one argument assume it's the path to the script which called main(), so run its tests.
+    elif argn == 1:
+        result = run_module(runner, module, split_path(sys.argv[0])[0], None)
```

Comment:
`split_path(sys.argv[0])`
This is to split out just the filename isn't it... I would think this will fail if you're running a unittest from a different folder?

Reply:
Yeah that's basically `os.path.dirname` . Not sure why that would fail, the point of the `path` argument is to add the script directory to `sys.path` right? So `python /path/to/test.py` is going to add `path/to`. I just realize this is not actually needed though, the directory of the executing script gets added to `sys.path` anyway. Moreover, as the comment says this is an assumption but thinking about it: this is just wrong and I'm going to remove it. If anything `argv[0]` could be used as prog name for argparse but that's it.